### ISSUE TITLE
java.lang.IllegalArgumentException: requirement failed: size=0 and step=0, but both must be positive

### ISSUE THREAD

**Issue by gaoli1990** [deeplearning4j/deeplearning4j#6853]
HI,
I was using deeplearning4j on spark. Currently I have the RDD data, which must be transferred to RDD[DataSet] used by dl4j. Each record of the RDD data is the vector and the label of each user. The following is my spark code, 
```
/**
    * transfer the Rdd data to the data used by dj4j
    * @param sparkSession
    * @param rddData: RDD[(userid, (user_vec, labels))]
    * @return
    */
def transferDataToDl4j(sparkSession: SparkSession, rddData: RDD[(String, (String, String))]) = {
    val ret = rddData
      .mapPartitions(iter => {
        iter
          .grouped(batchSize)
          .map(group => {
            val (u_vec, label) = group
              .map(row => {
                val (_, (u_vec_str, label_str)) = row
                val u_vec = u_vec_str.trim.split(",").map(e => e.toDouble)
                val label = label_str.trim.split(",").map(e => e.toDouble)
                (u_vec, label)
              })
              .unzip

            val dl4jUVec: INDArray = Nd4j.create(u_vec.toArray)
            val dl4jLabel: INDArray = Nd4j.create(label.toArray)
            val dl4jDataSet = new DataSet(dl4jUVec, dl4jLabel)
            dl4jDataSet
          })
      })
      .persist(StorageLevel.MEMORY_ONLY_SER)

    ret
  }
```
The following are the main part of network training.

// transfer the data to dj4j format
val trainingData = transferDataToDl4j(sparkSession, trainingDataPrepare)
    trainingDataPrepare.unpersist()

    // activate function config
    val activationFunction = activationFunc match {
      case "tanh" => Activation.TANH
      case "relu" => Activation.RELU
      case _ => Activation.LEAKYRELU
    }

    // create network configuration and conduct network training
    val conf: MultiLayerConfiguration = new NeuralNetConfiguration.Builder()
      .seed(seed)
      .weightInit(WeightInit.XAVIER)
      .updater(new Nesterovs(learningRate, 0.9))
      .l2(1e-4)
      .list()
      .layer(0, new DenseLayer.Builder().nIn(numInputs).nOut(numHiddenNodes)
                              .activation(activationFunction).build())
      .layer(1, new DenseLayer.Builder().nIn(numHiddenNodes).nOut(numHiddenNodes)
                              .activation(activationFunction).build())
      .layer(2, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                              .activation(Activation.SOFTMAX).nIn(numHiddenNodes).nOut(numOutputs).build)
      .pretrain(false).backprop(true)
      .build()

    // configuration for spark training
    val trainingMaster = new ParameterAveragingTrainingMaster.Builder(rddDataSetNumExamples)
      .averagingFrequency(averageFrequency)
      .workerPrefetchNumBatches(prefetchNumBatches)
      .batchSizePerWorker(batchSize)
      .saveUpdater(true)
      .build()

    // create the spark network
    val sparkNet = new SparkDl4jMultiLayer(sc, conf, trainingMaster)

    // executor training
    for(index <- 0 to nEpochs) {
      sparkNet.fit(trainingData)
      println(s"epoch completed: ${index} / ${nEpochs}")
    }

However, I got the error "java.lang.IllegalArgumentException: requirement failed: size=0 and step=0, but both must be positive", 
```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 224 in stage 15.0 failed 4 times, most recent failure: Lost task 224.3 in stage 15.0 (TID 1351, 9.10.19.86, executor 95): java.lang.IllegalArgumentException: requirement failed: size=0 and step=0, but both must be positive
	at scala.collection.Iterator$GroupedIterator.<init>(Iterator.scala:1023)
	at scala.collection.Iterator$class.grouped(Iterator.scala:1160)
	at scala.collection.AbstractIterator.grouped(Iterator.scala:1336)
	at com.tencent.omg.comment.mission.dl.algorithm.mlp.MLPTraining$$anonfun$8.apply(MLPTraining.scala:155)
	at com.tencent.omg.comment.mission.dl.algorithm.mlp.MLPTraining$$anonfun$8.apply(MLPTraining.scala:153)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:800)
	at org.apache.spark.rdd.RDD$$anonfun$mapPartitions$1$$anonfun$apply$23.apply(RDD.scala:800)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:337)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:335)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1109)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1083)
	at org.apache.spark.storage.BlockManager.doPut(BlockManager.scala:1018)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1083)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:809)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:335)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:286)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:87)
	at org.apache.spark.scheduler.Task.run(Task.scala:109)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:345)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

As I am the new beginner of dl4j and also the spark, hope to get help gratefully. 
Thanks!

**Comment by AlexDBlack** [comment]
Based on that stack trace, this doesn't look like it's related to DL4J - rather, it appears to be occurring in your own transferDataToDl4j method... Maybe test that in isolation first?
Beyond that, it's not clear what you are trying to do here (create training data, sure, but from what original format CSV or something?)

**Comment by gaoli1990** [comment]
> Based on that stack trace, this doesn't look like it's related to DL4J - rather, it appears to be occurring in your own transferDataToDl4j method... Maybe test that in isolation first?
> Beyond that, it's not clear what you are trying to do here (create training data, sure, but from what original format CSV or something?)

The original format is spark RDD data, with each record in the RDD is (userid, (user vector, labels)). The task is to using the user vector to predict the labels using MLP algorithm with dl4j.
